Asterix/Optical_System_functions.py

```diff
@@ -313,36 +329,46 @@ def EF_from_phase_and_ampl(self,
                                wavelength=None):
         """ --------------------------------------------------
         Create an electrical field from an phase and amplitude aberrations
-        
+
         Parameters
         ----------
         phase_abb : 2D array of size [self.dim_overpad_pupil, self.dim_overpad_pupil]. real
-            if 0, no phase aberration (default)  
-        
+            if 0, no phase aberration (default)
+
         phase_abb : 2D array of size [self.dim_overpad_pupil, self.dim_overpad_pupil]. real
-            if 0, no amplitude aberration (default)  
-            
+            if 0, no amplitude aberration (default)
+
 
         wavelength : float. Default is the reference self.wavelength_0
-             current wavelength in m. 
+             current wavelength in m.
 
 
         Returns
         ------
-        EF : 2D array, of size [self.dim_overpad_pupil, self.dim_overpad_pupil] 
+        EF : 2D array, of size [self.dim_overpad_pupil, self.dim_overpad_pupil]
             Electric field in the pupil plane a the exit of the system
-        
+
         AUTHOR : Johan Mazoyer
         """
         if np.iscomplexobj(phase_abb) or np.iscomplexobj(ampl_abb):
             raise Exception(
                 "phase_abb and ampl_abb must be real arrays or float, not complex"
             )
 
-        if (phase_abb == 0.).all() and (ampl_abb == 0).all():
+        if isinstance(phase_abb, (int, float, np.float)):
+            phase_abb = np.full(
+                (self.dim_overpad_pupil, self.dim_overpad_pupil),
+                np.float(phase_abb))
```

Comment:
What does this do?

Reply:
if you put phase_abb = 0. or  phase_abb = 0 it transform to np.zeros(self.dim_overpad_pupil, self.dim_overpad_pupil)

also works for any constant (e.g. phase_abb = np.pi)

Reply:
ok but do we really need this? exp(i phase) multiplies a 2D-array. If phase_abb = Cte then, pup = pup0*exp(i phase_abb) works. No need to create and 2D-array with constant value for all pixels (will same memory).

Reply:
Yes, you're perfectly right, we can easily save time! this will be done today